zio-config/src/main/scala/zio/config/Sources.scala

```diff
@@ -1,17 +1,39 @@
 package zio.config
 
-import zio.{ IO, Task }
+import zio.system.System
+import zio.{ Task, ZIO }
 
 trait Sources {
-  def envSource: Task[ConfigSource] =
-    Task(sys.env).map(mapSource)
+  def envSource: ZIO[System, Throwable, ConfigSource] =
+    ZIO.access { env =>
+      new ConfigSource {
+        val configService: ConfigSource.Service =
+          (path: String) =>
+            env.system.env(path).flatMap { (oStr: Option[String]) =>
+              oStr.fold[Task[String]](
+                ZIO.fail(new IllegalArgumentException(s"Not defined in system environment: $path"))
+              )(ZIO.succeed)
+            }
+      }
+    }
 
-  def propSource: Task[ConfigSource] =
-    Task(sys.props.toMap).map(mapSource)
+  def propSource: ZIO[System, Throwable, ConfigSource] =
+    ZIO.access { env =>
+      new ConfigSource {
+        val configService: ConfigSource.Service =
+          (path: String) =>
+            env.system.property(path).flatMap { (oStr: Option[String]) =>
+              oStr.fold[Task[String]](
+                ZIO.fail(new IllegalArgumentException(s"System property not found for key: $path"))
```

Comment:
@Michaelt293 

1. (s"System property not found for key: $path")). How is this different to MissingValue handling done in `Read.scala`. Could you take a look at it. You may have some inputs there.

```scala
            value <- config
                      .getConfigValue(path)
                      .mapError(_ => ReadErrors(MissingValue(path)))
```

2. This might be a silly question, but just to be on the same page. If we have `n` keys to read, then we call `sys.env(key)` for sure. And you are saying that's ok, because `sys.properties` may faill due to security, but `sys.property(key)` won't . Is that right?

Reply:
The only minor issue here, is 
```
           value <- config
                      .getConfigValue(path)
                      .mapError(_ => ReadErrors(MissingValue(path)))
```
We are mapping all errors `getConfigValue` which is now returning a `Task`, to be `MissingValue`.
Is there way you propagate this information. That is, ensuring that the failure in `Task` is actually due to a `MissingValue` and hence tagging it?
`.mapOnlyMissingValueError(error => MissingValue)`

Reply:
And try running an example giving `sys.env`. We don't have a test which actually test system environment.

Reply:
I think if this is in place, its good to merge 👍

Reply:
By doing thee above changes I mentioned, we get rid of illegal argument exceptions in the code

Reply:
Good points, I have used `ReadError` as the error type now. I have also added a `cause` field to `MissingValue` to provide more information on what went wrong.  I have also added a `FatalError` case to the `ReadError` type which ends up handling the `SecurityException`  that can occur when accessing environment and system properties.

 Regarding `sys.properties` vs `sys.property(key)`, `sys.property(key)` can still throw a `SecurityException`. My understanding is that things can be configured so that `sys.properties` could throw an exception and `sys.property(key)` returns a value. This is probably and edge case that you would typically would not encounter.

Reply:
Regarding testing, I have not manually tested these changes. Perhaps we should first try  this before merging this PR.